python_modules/dagster/dagster/_config/pythonic_config/config.py

```diff
@@ -342,29 +390,33 @@ def infer_schema_from_config_class(
     )
 
     fields: Dict[str, Field] = {}
-    for pydantic_field in model_cls.__fields__.values():
-        if pydantic_field.name not in fields_to_omit:
-            if isinstance(pydantic_field.default, Field):
+    for key, pydantic_field_info in model_fields(model_cls).items():
+        if _is_annotated_as_resource_type(
+            pydantic_field_info.annotation, pydantic_field_info.metadata
+        ):
+            continue
+
+        alias = pydantic_field_info.alias if pydantic_field_info.alias else key
+        if key not in fields_to_omit:
+            if isinstance(pydantic_field_info.default, Field):
```

Comment:
I'd recommend importing `Field` as an alias like `DagsterField` to make this super explicit